.github/workflows/build.yml

```diff
@@ -0,0 +1,83 @@
+name: Docker build
+
+on: [pull_request, push]
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+
+    env:
+      release: "2021.3.0"
+      defaultpython: "3.8"
+
+    strategy:
+      matrix:
+        python: ["3.7", "3.8", "3.9"]
+        image:
+          - tag: "daskdev/dask"
+            context: "./base"
+          - tag: "daskdev/notebook"
+            context: "./notebook"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        if: github.repository == 'dask/dask-docker' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Generate tags
+        id: tags
+        env:
+          image: ${{ matrix.image.tag }}
+          python: ${{ matrix.python }}
+        run: |
+          tags="${image}:${release}-py${python}"
+          if [ "$python" == "$defaultpython" ]; then
+            tags=${image}:latest,${image}:${release},$tags
+          fi
+          echo "::set-output name=tags::${tags}"
+
+      - name: Checkout upstream
+        uses: actions/checkout@v2
+        if: matrix.image.tag == 'daskdev/notebook'
+        with:
+          repository: jupyter/docker-stacks
+          ref: master
+          path: docker-stacks
+
+      - name: Build upstream Jupyter Lab image
+        uses: docker/build-push-action@v2
+        if: matrix.image.tag == 'daskdev/notebook'
+        with:
+          context: ./docker-stacks/base-notebook
+          push: false
+          load: true
+          platforms: linux/amd64
+          tags: jupyter/base-notebook:lab
```

Comment:
@jrbourbeau The image is being built here.

If we use the official one we can't specify the Python version because that has to be configured at build time.

Reply:
Gotcha -- I was trying to resolve the CI failures here https://github.com/dask/dask-docker/runs/2143679297

Reply:
Hrm strange. I wonder if buildx requires you to be fully explicit with the tag name.